cmd/data.go

```diff
@@ -0,0 +1,147 @@
+package cmd
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"path"
+	"time"
+
+	"github.com/cloudspannerecosystem/harbourbridge/conversion"
+	"github.com/cloudspannerecosystem/harbourbridge/internal"
+	"github.com/google/subcommands"
+)
+
+// DataCmd struct with flags.
+type DataCmd struct {
+	source          string
+	sourceProfile   string
+	target          string
+	targetProfile   string
+	skipForeignKeys bool
+	sessionJSON     string
+	filePrefix      string // TODO: move filePrefix to global flags
+}
+
+// Name returns the name of operation.
+func (cmd *DataCmd) Name() string {
+	return "data"
+}
+
+// Synopsis returns summary of operation.
+func (cmd *DataCmd) Synopsis() string {
+	return "migrate data from source db to target db"
+}
+
+// Usage returns usage info of the command.
+func (cmd *DataCmd) Usage() string {
+	return fmt.Sprintf(`%v data -session=[session_file] -source=[source] -target-profile="instance=my-instance"...
+
+Migrate data from source db to target db. Source db dump file can be specified
+by either file param in source-profile or piped to stdin. Connection profile
+for source databases in direct connect mode can be specified by setting
+appropriate environment variables. The data flags are:
+`, path.Base(os.Args[0]))
+}
+
+// SetFlags sets the flags.
+func (cmd *DataCmd) SetFlags(f *flag.FlagSet) {
+	f.StringVar(&cmd.source, "source", "", "Flag for specifying source DB, (e.g., `PostgreSQL`, `MySQL`, `DynamoDB`)")
+	f.StringVar(&cmd.sourceProfile, "source-profile", "", "Flag for specifying connection profile for source database e.g., \"file=<path>,format=dump\"")
+	f.StringVar(&cmd.sessionJSON, "session", "", "Specifies the file we restore session state from")
+	f.StringVar(&cmd.target, "target", "Spanner", "Specifies the target DB, defaults to Spanner (accepted values: `Spanner`)")
+	f.StringVar(&cmd.targetProfile, "target-profile", "", "Flag for specifying connection profile for target database e.g., \"dialect=postgresql\"")
+	flag.BoolVar(&cmd.skipForeignKeys, "skip-foreign-keys", false, "Skip creating foreign keys after data migration is complete (ddl statements for foreign keys can still be found in the downloaded schema.ddl.txt file and the same can be applied separately)")
+	f.StringVar(&cmd.filePrefix, "prefix", "", "File prefix for generated files")
+}
+
+func (cmd *DataCmd) Execute(ctx context.Context, f *flag.FlagSet, _ ...interface{}) subcommands.ExitStatus {
+	var err error
+	defer func() {
+		if err != nil {
+			fmt.Printf("FATAL error: %v\n", err)
+		}
```

Comment:
I think we can later refactor these parts into schema specific changes and data specific, and call them from Execute() of schema, data and eval modes since there is much repetition across these modes.